css/src/atomics/spacing.scss

```diff
@@ -34,8 +34,30 @@ $layout-sizes: (
 	.#{$paddingKey}#{$separator}#{$sizeKey} {
 		padding: sizeValue($sizeKey, $sizeValue) !important;
 	}
+}
 
-	// margin-inline and margin-block property isn't supported by IE, Edge, and Safari yet (as of Apr 2021)
+@each $position-key, $direction in $directions {
```

Comment:
It may be that I did this work too quickly, but could you take the following idea and see if it can work. The keys with it are

1. We don't need to do anything super special to deal with sides or -inline, just make a list of the properties we want to use.
2. Iterpolate the property into the name and the rule.

```scss
$separator: '-';

$spacing-properties: 'margin', 'margin-inline', 'margin-block', 'margin-top', 'margin-bottom',
	'margin-left', 'margin-right', 'padding', 'padding-inline', 'padding-block', 'padding-top',
	'padding-bottom', 'padding-left', 'padding-right' !default;

$layout-sizes: (
	('xs', $layout-1),
	('s', $layout-2),
	('m', $layout-3),
	('l', $layout-4),
	('xl', $layout-5),
	('xxl', $layout-6),
	('xxxl', $layout-7),
	('none', $layout-0)
);

@function sizeValue($key, $value) {
	@return if($key == 'none', 0, $value);
}

@each $property in $spacing-properties {
	// interates through margin, margin-inline, margin-block, margin-inline
	@each $size in $layout-sizes {
		// iterates over values 'xs', 's', 'l' ... 'none'
		$sizeKey: nth($size, 1);
		$sizeValue: nth($size, 2);

		// .<property>-<value>
		.#{$property}#{$separator}#{$size} {
			#{$property}: $sizeValue !important;
		}
		// margin-xs, margin-s, margin-m ... margin-none ... etc
	}
}

@include tablet {
	@each $property in $spacing-properties {
		// interates through margin, margin-inline, margin-block, margin-inline
		@each $size in $layout-sizes {
			// iterates over values 'xs', 's', 'l' ... 'none'
			$sizeKey: nth($size, 1);
			$sizeValue: nth($size, 2);
	
			// .<property>-<value>
			.#{$property}#{$separator}#{$size}-tablet { // 🌟 only change
				#{$property}: $sizeValue !important;
			}
			// margin-xs, margin-s, margin-m ... margin-none ... etc
		}
	}
}

```

Reply:
Updated